R/premium.R

```diff
@@ -111,147 +76,103 @@ format_from_to_date <- function(x = NULL) {
-#' }
-#'
-#' @section Geospatial:
-#' \itemize{
-#'   \item **bounding_box: \[west_long south_lat east_long north_lat\]** ~~ lat/long coordinates box
-#'   \item **point_radius: \[lon lat radius\]** ~~ center of search radius
-#'   \item **has:geo**           ~~ uses geotagging
-#'   \item **place:**            ~~ by place
-#'   \item **place_country:**    ~~ by country
-#'   \item **has:profile_geo**   ~~ geo associated with profile
-#'   \item **profile_country:**  ~~ country associated with profile
-#'   \item **profile_region:**   ~~ region associated with profile
-#'   \item **profile_locality:** ~~ locality associated with profile
-#' }
-#'
-#' @return A tibble data frame of Twitter data
-#' @examples
-#'
-#' \dontrun{
-#' ## format datetime for one week ago
-#' toDate <- format(Sys.time() - 60 * 60 * 24 * 7, "%Y%m%d%H%M")
-#'
-#' ## search 30day for up to 300 rstats tweets sent before the last week
-#' rt <- search_30day("#rstats", n = 300,
-#'   env_name = "research", toDate = toDate)
-#' }
-#'
+#' @rdname search_fullarchive
 #' @export
 search_30day <- function(q, n = 100, fromDate = NULL, toDate = NULL,
                           env_name = NULL, safedir = NULL,
                           parse = TRUE,
                           token = NULL) {
-  token <- check_token(token)
-  if (!length(get_app_secret(token))) {
-    stop(paste0("This token does not have an app secret and therefore cannot ",
-      "create a bearer token"), call. = FALSE)
-  }
-  token <- bearer_token(token)
-  if (is.null(safedir) || isTRUE(safedir)) {
-    safedir <- paste0("premium-", format(Sys.Date(), "%Y%m%d"))
-  } else if (isFALSE(safedir)) {
-    safedir <- NULL
-  } else {
-    safedir <- safedir
-  }
-  fromDate <- format_from_to_date(fromDate)
-  toDate <- format_from_to_date(toDate)
-  if (is.null(env_name)) {
-    stop("Must provide dev environment name")
-  }
-  r <- search_tweets(q,
+  
+  search_premium("30day", 
+    q = q, 
+    n = n,
     fromDate = fromDate,
     toDate = toDate,
-    premium = premium_api("30day", env_name),
-    parse = FALSE, n = n,
+    env_name = env_name,
     safedir = safedir,
-    token = token)
+    parse = parse,
+    token = token
+  )
+}
+
+
+search_premium <- function(product, q, n = 100, fromDate = NULL, toDate = NULL,
+                          env_name = NULL, safedir = NULL,
+                          parse = TRUE,
+                          token = NULL) {
+  
+  if (is.null(env_name)) {
+    stop("Must provide dev environment name")
+  }
+  if (!is.null(safedir)) {
+    stop("`safedir` temporarily not supported")
+  }
+  
+  token <- bearer_token(token)
+  params <- search_params(q, 
+    n = n,
+    fromDate = format_from_to_date(fromDate),
+    toDate = format_from_to_date(toDate),
+    max_id = max_id,
+  )
+  
+  api <- paste0("search/tweets/", product, "/", env_name)
+  result <- TWIT_paginate_max_id(token, api, params,
+    get_max_id = function(x) x$statuses$id_str,
+    n = n,
+    page_size = 100,
```

Comment:
In permium search isn't it the maxResults parameter used instead of page_size? Also it is 500 the maximum: https://developer.twitter.com/en/docs/twitter-api/premium/search-api/api-reference/premium-search#Pagination
Several users complained that their queries using rtweet became more expensive due to not  fully use the API. This also applies to fullarchive.

Reply:
Good catch, I'll fix.